spec/zoneddatetime.html

```diff
@@ -0,0 +1,1049 @@
+      <h1>Temporal.ZonedDateTime.prototype.getISOFields ( )</h1>
+      <p>
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _zonedDateTime_ be the *this* value.
+        1. Perform ? RequireInternalSlot(_zonedDateTime_, [[InitializedTemporalZonedDateTime]]).
+        1. Let _fields_ be ? ObjectCreate(%ObjectPrototype%).
+        1. <mark>TODO</mark>.
+        1. Perform ! CreateDataPropertyOrThrow(_fields_, *"timeZone"*, _zonedDateTime_.[[TimeZone]]).
+        1. Perform ! CreateDataPropertyOrThrow(_fields_, *"calendar"*, _zonedDateTime_.[[Calendar]]).
+        1. Return _fields_.
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-properties-of-temporal-zoneddatetime-instances">
+    <h1>Properties of Temporal.ZonedDateTime Instances</h1>
+    <p>
+      Temporal.ZonedDateTime instances are ordinary objects that inherit properties from the %Temporal.ZonedDateTime.prototype% intrinsic object.
+      Temporal.ZonedDateTime instances are initially created with the internal slots described in <emu-xref href="#table-internal-slots-of-temporalzoneddatetime-instances"></emu-xref>.
+    </p>
+    <emu-table id="table-internal-slots-of-temporalzoneddatetime-instances" caption="Internal Slots of Temporal.ZonedDateTime Instances">
+      <table>
+        <tbody>
+          <tr>
+            <th>
+              Internal Slot
+            </th>
+            <th>
+              Description
+            </th>
+          </tr>
+          <tr>
+            <td>
+              [[InitializedTemporalZonedDateTime]]
+            </td>
+            <td>
+              The only specified use of this slot is for distinguishing Temporal.ZonedDateTime instances from other objects.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[Nanoseconds]]
+            </td>
+            <td>
+              A BigInt value representing the number of nanoseconds since the Unix epoch.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[TimeZone]]
+            </td>
+            <td>
+              An Object representing the time zone.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[Calendar]]
+            </td>
+            <td>
+              An Object representing the calendar.
+            </td>
+          </tr>
+        </tbody>
+      </table>
+    </emu-table>
+  </emu-clause>
+
+  <emu-clause id="sec-temporal-zoneddatetime-abstract-ops">
+    <h1>Abstract operations</h1>
+
+    <emu-clause id="sec-temporal-totemporalzoneddatetime" aoid="ToTemporalZonedDateTime">
+      <h1>ToTemporalZonedDateTime ( _item_ [ , _constructor_ [ , _overflow_, _disambiguation_, _offset_ ] ] )</h1>
+      <p>
+        The abstract operation ToTemporalZonedDateTime returns its argument _item_ if it is already a Temporal.ZonedDateTime instance, converts _item_ to a new Temporal.Date instance if possible, and throws otherwise.
```

Comment:
I thought that `from()` was always guaranteed to create a new instance, and to clone the instance if it's already an instance of the same type?

Reply:
Oh never mind. I think I understand: this AO is used for internal conversions, while a different AO (which does clone) is used for `from()`?

Reply:
That's right. In from() we check if the object is an instance, and clone and return it if so. If it's already not an instance, then we use this AO.